### PR TITLE
Wrote out some conventions we've acquired over the past few weeks

### PR DESCRIPTION
**Pull request by jasonrhodes**
I might be missing things and these descriptions might not be 100% clear so feel free to suggest changes/additions

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,39 +2,60 @@
...
-## Docs
+## About the project
+
+Before jumping into writing code, here are a few things you'll want to be aware of:
+
+1. We use ESLint to enforce automated linting. The config is extended from a few places and defined inside the package.json file, under the "eslintConfig" key (instead of using a root level .eslintrc file).
+    * To run our linting, you can run `npm run lint` or `npm run lint -- --fix` to run in "auto fix mode". 
+    * Linting is run during the build process but NOT during tests (jest runs tests in watch mode)
+    * We are always looking for ways to move things into automated linting instead of just having "conventions" listed later in this section.
+    * We should consider looking at some combination of prettier and eslint at some point.
```
Comment (tough-griff) [inline README.md]: :+1:

File: README.md [hunk 2]
```diff
@@ -2,39 +2,60 @@
...
+
+1. **File and folder names:** everything is camelCase except for component files and other class-only files, which are PascalCase.
+
+1. **index.js "directory" files:** when there are 3 or more components in a sub-directory (e.g. src/pages/webhooks), export all of them inside of an index.js file there, for easier and less verbose imports elsewhere.
+
+1. **Sub-directory groups:** Any time you have more than one file making up a component or related set of components, put them in a sub-directory (e.g. src/components/collection/*)
+
+1. **Magic module resolution:** we use webpack tomfoolery and jest shenanigans to make our modules resolve relative to `src/`, so you can `import something from whatever/yeah` so long as there is a `src/whatever/yeah.js` file. And so on. You should prefer this style to relative imports unless the files are directly next to each other or are related to each other and no more than one directory away.
```
Comment (tough-griff) [inline README.md]: Let's re-open the discussion about changing this to `import src/whatever/yeah`

File: README.md [hunk 3]
```diff
@@ -2,39 +2,60 @@
...
+
+Before jumping into writing code, here are a few things you'll want to be aware of:
+
+1. We use ESLint to enforce automated linting. The config is extended from a few places and defined inside the package.json file, under the "eslintConfig" key (instead of using a root level .eslintrc file).
+    * To run our linting, you can run `npm run lint` or `npm run lint -- --fix` to run in "auto fix mode". 
+    * Linting is run during the build process but NOT during tests (jest runs tests in watch mode)
+    * We are always looking for ways to move things into automated linting instead of just having "conventions" listed later in this section.
+    * We should consider looking at some combination of prettier and eslint at some point.
+
+1. **File and folder names:** everything is camelCase except for component files and other class-only files, which are PascalCase.
```
Comment (tough-griff) [inline README.md]: note: file names should match the default export

File: README.md [hunk 4]
```diff
@@ -2,39 +2,60 @@
...
+
+1. **index.js "directory" files:** when there are 3 or more components in a sub-directory (e.g. src/pages/webhooks), export all of them inside of an index.js file there, for easier and less verbose imports elsewhere.
+
+1. **Sub-directory groups:** Any time you have more than one file making up a component or related set of components, put them in a sub-directory (e.g. src/components/collection/*)
+
+1. **Magic module resolution:** we use webpack tomfoolery and jest shenanigans to make our modules resolve relative to `src/`, so you can `import something from whatever/yeah` so long as there is a `src/whatever/yeah.js` file. And so on. You should prefer this style to relative imports unless the files are directly next to each other or are related to each other and no more than one directory away.
+
+### Testing Conventions
```
Comment (tough-griff) [inline README.md]: Maybe add a note about using built in assertions  whenever possible? The error messages are much nicer that way.

i. e. 
```js
// use this
expect(fn).toHaveBeenCalledWith(arg1, arg2);
// not this
expect(fn.mock.calls[0]).toEqual([arg1, arg2]);
```

Comment (tough-griff) [reply README.md]: maybe `docs/jest-mocking` should just be `docs/jest`

File: README.md (hunk 2 above)
Comment (jonambas) [reply README.md]: Oh yeah, requiring `src` in the path is probably the way to go

Comment (jasonrhodes) [reply README.md]: That's cool -- when we do this change soon we can update this doc.

File: README.md (hunk 4 above)
Comment (jasonrhodes) [reply README.md]: Might need both, but yeah we'll see.

**Pull request merged**